solutions/ac-GCDOnBlackboard.mdx

```diff
@@ -0,0 +1,69 @@
+---
+id: ac-GCDOnBlackboard
+source: AC
+title: GCD On Blackboard
+author: Andrew Cheng
+---
+
+## Time Complexity
+$\mathcal{O}(N \log N)$ (GCD operations take $\mathcal{O}(\log N)$ time)
+
+## Solution
+The question asks us to find the maximum possible GCD of the remaining $N-1$ numbers
+after taking any one of them away. Naively trying every single combination will result in a
+complexity of $\mathcal{O}(N^2)$ as there are $N$ different ways to take away a number and calculating
+the GCD will take $\mathcal{O}(N)$ time per case. 
+
+To speed this process up, we can utilize a prefix sum and a suffix sum. Let $l[i]$ = $\gcd_{j=1}^{i} a[j]$ and $r[i]$ = $\gcd_{j=i}^{N} a[j]$.
+Then the answer is the maximum of $\gcd(l[i-1],r[i+1]),$ $i \in [2,N-1]$. Though remember to check for the case
+where the first or the last number is replaced (line 25).
```

Comment:
maybe add a comment instead of referring to line 25 (since suggestions might change its position)